### PR TITLE
feat/add pinia store

### LINKED ISSUE #45: RFC: Add a store (e.g., pinia) to the starter
### Describe the feature

Requested by @MaloLebrin here: https://github.com/sidestream-tech/sidebase/discussions/44 they suggested that adding a store, more specifically pinia, would be good.

I agree that it would be helpful, especially with getting typing etc right during the setup can be a bit of a hassle.

### Additional information

We should checkout other storage solutions (e.g.: nuxt already has `useSate` which allows for small footprint stored in a very easy to use manner).

### PR DESCRIPTION
**Pull request by MaloLebrin**
Closes #45 

Checklist:
- [x] issue number linked above after pound (`#`)
    - replace "Closes " with "Contributes to" or other if this PR does not close the issue
- [x] issue checkboxes are all addressed
- [x] manually checked my feature / not applicable
- [x] wrote tests / not applicable
- [x] attached screenshots / not applicable

### CODE REVIEW

File: store/exempleStore.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export const useExempleStore = defineStore('counter', {
```
Comment (zoey-kaiser) [inline store/exempleStore.ts:1]:
```suggestion
export const useExampleStore = defineStore('counter', {
```

File: pages/index.vue [hunk 2]
```diff
@@ -25,7 +27,7 @@ const ratingValue = ref(5)
     <div class="flex w-4/6 max-w-lg flex-col">
       <SidebaseLogo class="w-72 fill-blue-200" />
       <h1 class="text-xl">
-        Welcome to <a href="https://github.com/sidestream-tech/sidebase" target="_blank">sidebase</a>!
+        Welcome to <a href="https://github.com/sidestream-tech/sidebase" target="_blank">sidebase {{ store.doubleCount }}</a>!
```
Comment (zoey-kaiser) [inline pages/index.vue:30]: Can we maybe show a "real-world" example for the store?
My suggestion would be to integrate the `a-rate` (where the user can set a star rating) with the store. 

Alternatively you can also add a button which increases the count (you can also display the count inside of the button) 😄

File: store/exempleStore.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  state: () => ({ count: 0, name: 'Eduardo' }),
```
Comment (zoey-kaiser) [inline store/exempleStore.ts:2]: If you add names or other default values, please just fill them with something not related to a specific person. 

If you want to add a input field where users can define a name, I would recommend to fill it with "sidebase" 🤗

File: pages/index.vue [hunk 4]
```diff
@@ -11,6 +11,8 @@ onMounted(() => {
   refreshDataInterval = setInterval(refresh, 2000)
 })
 
+const store = useExempleStore()
```
Comment (zoey-kaiser) [inline pages/index.vue:14]:
```suggestion
const store = useExampleStore()
```

**Review by zoey-kaiser — CHANGES_REQUESTED** [review]
Hi @MaloLebrin! 👋 

Thank you so much for your pull request! It means a lot to us to see members of the community take an interest in the development of `sidebase`!

I am very happy with how you added pina to sidebase so most of my review points are very nit-picky, to add some more polish. 
Have a look through my suggestions and feel free to disagree with any points I have made! 😄 

---

Aside from the points I made during the review, I believe we also need to add some points to our documentation before we merge this PR, however we can also take care of writing these (if you don't feel like working on it). 

- [ ] Update ReadMe to include `pina` in our list of included packages
- [ ] Write a quick tutorial on how to use `pina` (maybe also with some more links to help developers if they run into any issues)
- [ ] Update texts on `sidebase` website (this is for us 😆)

**Comment by zoey-kaiser** [comment]
While doing some more testing I ran into a severe issue. When using the store in `historie` I get the following error:

```
Error while collecting story /sidebase/components/example/Status.story.vue:
TypeError [ERR_PACKAGE_IMPORT_NOT_DEFINED]: Package import specifier "#imports" is not defined in package /sidebase/node_modules/@pinia/nuxt/package.json imported from /sidebase/node_modules/@pinia/nuxt/dist/runtime/composables.mjs
    at new NodeError (node:internal/errors:371:5)
    at throwImportNotDefined (node:internal/modules/esm/resolve:442:9)
    at packageImportsResolve (node:internal/modules/esm/resolve:819:3)
    at moduleResolve (node:internal/modules/esm/resolve:973:21)
    at defaultResolve (node:internal/modules/esm/resolve:1080:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:530:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:251:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:79:40)
    at link (node:internal/modules/esm/module_job:78:36)
```

Have you looked into how to correctly integrate pina into historie?

File: store/exempleStore.ts (hunk 3 above)
Comment (MaloLebrin) [reply store/exempleStore.ts:2]: i simply removed field "name"

**Comment by MaloLebrin** [comment]
> While doing some more testing I ran into a severe issue. When using the store in `historie` I get the following error:
> 
> ```
> Error while collecting story /sidebase/components/example/Status.story.vue:
> TypeError [ERR_PACKAGE_IMPORT_NOT_DEFINED]: Package import specifier "#imports" is not defined in package /sidebase/node_modules/@pinia/nuxt/package.json imported from /sidebase/node_modules/@pinia/nuxt/dist/runtime/composables.mjs
>     at new NodeError (node:internal/errors:371:5)
>     at throwImportNotDefined (node:internal/modules/esm/resolve:442:9)
>     at packageImportsResolve (node:internal/modules/esm/resolve:819:3)
>     at moduleResolve (node:internal/modules/esm/resolve:973:21)
>     at defaultResolve (node:internal/modules/esm/resolve:1080:11)
>     at ESMLoader.resolve (node:internal/modules/esm/loader:530:30)
>     at ESMLoader.getModuleJob (node:internal/modules/esm/loader:251:18)
>     at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:79:40)
>     at link (node:internal/modules/esm/module_job:78:36)
> ```
> 
> Have you looked into how to correctly integrate pina into historie?



> While doing some more testing I ran into a severe issue. When using the store in `historie` I get the following error:
> 
> ```
> Error while collecting story /sidebase/components/example/Status.story.vue:
> TypeError [ERR_PACKAGE_IMPORT_NOT_DEFINED]: Package import specifier "#imports" is not defined in package /sidebase/node_modules/@pinia/nuxt/package.json imported from /sidebase/node_modules/@pinia/nuxt/dist/runtime/composables.mjs
>     at new NodeError (node:internal/errors:371:5)
>     at throwImportNotDefined (node:internal/modules/esm/resolve:442:9)
>     at packageImportsResolve (node:internal/modules/esm/resolve:819:3)
>     at moduleResolve (node:internal/modules/esm/resolve:973:21)
>     at defaultResolve (node:internal/modules/esm/resolve:1080:11)
>     at ESMLoader.resolve (node:internal/modules/esm/loader:530:30)
>     at ESMLoader.getModuleJob (node:internal/modules/esm/loader:251:18)
>     at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:79:40)
>     at link (node:internal/modules/esm/module_job:78:36)
> ```
> 
> Have you looked into how to correctly integrate pina into historie?

Did you import manualy the store ? I didn't use histoire before, but i think it's because of autoimport i've made. But you don't have to autoimport every store you can import it manualy

**Comment by zoey-kaiser** [comment]
> After some test i think the problem in with pinia in histoire. I didn't use histoire before sorry

No problem! I will do some research into it tomorrow and see if I can find any leads. Thank you for the amazing start ❤️

**Comment by MaloLebrin** [comment]
> While doing some more testing I ran into a severe issue. When using the store in `historie` I get the following error:
> 
> ```
> Error while collecting story /sidebase/components/example/Status.story.vue:
> TypeError [ERR_PACKAGE_IMPORT_NOT_DEFINED]: Package import specifier "#imports" is not defined in package /sidebase/node_modules/@pinia/nuxt/package.json imported from /sidebase/node_modules/@pinia/nuxt/dist/runtime/composables.mjs
>     at new NodeError (node:internal/errors:371:5)
>     at throwImportNotDefined (node:internal/modules/esm/resolve:442:9)
>     at packageImportsResolve (node:internal/modules/esm/resolve:819:3)
>     at moduleResolve (node:internal/modules/esm/resolve:973:21)
>     at defaultResolve (node:internal/modules/esm/resolve:1080:11)
>     at ESMLoader.resolve (node:internal/modules/esm/loader:530:30)
>     at ESMLoader.getModuleJob (node:internal/modules/esm/loader:251:18)
>     at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:79:40)
>     at link (node:internal/modules/esm/module_job:78:36)
> ```
> 
> Have you looked into how to correctly integrate pina into historie?



> Hi @MaloLebrin! 👋
> 
> Thank you so much for your pull request! It means a lot to us to see members of the community take an interest in the development of `sidebase`!
> 
> I am very happy with how you added pina to sidebase so most of my review points are very nit-picky, to add some more polish. Have a look through my suggestions and feel free to disagree with any points I have made! 😄
> 
> Aside from the points I made during the review, I believe we also need to add some points to our documentation before we merge this PR, however we can also take care of writing these (if you don't feel like working on it).
> 
>     * [ ]  Update ReadMe to include `pina` in our list of included packages
> 
>     * [ ]  Write a quick tutorial on how to use `pina` (maybe also with some more links to help developers if they run into any issues)
> 
>     * [ ]  Update texts on `sidebase` website (this is for us 😆)

what text are you talking about in third task ?

**Comment by zoey-kaiser** [comment]
> what text are you talking about in third task ?

I am referring to the texts on [sidebase.io/why](https://sidebase.io/why), it was a reminder for myself to update the texts once we merged! 😄

**Comment by zoey-kaiser** [comment]
I opened an issue to hopefully get more infos on how to best handle the integration:

https://github.com/histoire-dev/histoire/issues/303

**Comment by MaloLebrin** [comment]
> 

Hello, I think the requested changes are made now, what do you think?

**Comment by zoey-kaiser** [comment]
Hi @MaloLebrin,

I am currently still waiting on update from https://github.com/histoire-dev/histoire/issues/286 to our issue with historie. I will re-review once we get an answer!

**Comment by zoey-kaiser** [comment]
Hi @MaloLebrin!

Sorry for the long delay in this PR. We are currently in the process of removing historie from sidebase, as it is still causing issues with the newer Nuxt versions and other packages such as Pinia (#75)! 

I will close this PR for now. If you still want to add pinia you are welcome to do so after we merge #75. Otherwise I would also be willing to create a new PR and add it myself afterwards! Let me know which option you prefer!

Thank you for your PR and request!

**Comment by MaloLebrin** [comment]
> 

I think we shloud rebase from new version after merge #75 and merge this PR ?

**Comment by zoey-kaiser** [comment]
> I think we shloud rebase from new version after merge #75 and merge this PR ?

Agreed! Ill reopen this once we did it! We are having some build issues with the newest Nuxt versions and are still trying to figure it out 😢

**Pull request closed without merging**